components/nimbus/src/lib.rs

```diff
@@ -488,6 +495,25 @@ impl NimbusClient {
         let helper = NimbusStringHelper::new(context.as_object().unwrap().to_owned());
         Ok(Arc::new(helper))
     }
+
+    pub fn record_exposure_event(&self, feature_id: String) -> Result<()> {
+        // Next, we search for any experiment that has a matching featureId. This depends on the
+        // fact that we can only be enrolled in a single experiment per feature, so there should
+        // only ever be zero or one experiments for a given featureId.
+        if let Some(exp) = self
+            .get_active_experiments()?
+            .iter()
+            .find(|exp| exp.feature_ids.contains(&feature_id))
```

Comment:
I feel sure we can do something a bit more efficient than this, but I can't think of a better way that's as non-invasive as this.

Reply:
Yes, hoping @tarikeshaq might have some ideas on this. It was as close to the logic of what we were doing in Swift as I could arrive at.

Reply:
Yeah I don't have any brilliant ideas here 😞 We can modify our state to include a map from feature_id -> experiment maybe? but that's a little intrusive and we'll need to be careful keeping it up to date as experiments change

Regardless I'm happy with this, and we can always improve it if we recognize that the time spent iterating over experiments is too expensive

Reply:
> We can modify our state to include a map from feature_id -> experiment maybe? but that's a little intrusive and we'll need to be careful keeping it up to date as experiments change
Yes, we do keep a map of `feature_id` to `EnrolledFeatureConfig`, which could keep an optional experiment id.

It's kept in memory and updated everytime `apply_pending_experiments` is called.

> Regardless I'm happy with this, and we can always improve it if we recognize that the time spent iterating over experiments is too expensive

Agreed.